CRM/Activity/Tokens.php

```diff
@@ -162,6 +148,29 @@ public function getBasicTokens(): array {
     return $this->basicTokens;
   }
 
+  /**
+   * Get tokens that are special or calculated for this enitty.
+   *
+   * @return array|array[]
+   */
+  protected function getBespokeTokens(): array {
+    $tokens = [];
+    if (array_key_exists('CiviCase', CRM_Core_Component::getEnabledComponents())) {
+      $tokens['case_id'] = ts('Activity Case ID');
+      return [
+        'case_id' => [
```

Comment:
This token exists because it was added in the past but there are 1-1 entity validilty issues

Reply:
Right, at best it could get you *a* case id (but no guarantee which one)